tests/test_all_rules.py

```diff
@@ -641,6 +641,39 @@ def test_deprecated_rules_modified(self):
         if result:
             self.fail(f"Deprecated rules {result} has been modified")
 
+    @unittest.skipIf(os.getenv('GITHUB_EVENT_NAME') == 'push',
+                     "Skipping this test when not running on pull requests.")
```

Comment:
Only concern I had with this is that we won't be able to naturally run unit tests without setting this environment variable. Probably not ideal for people (internally or the community) that want to test everything locally before opening the PR.

Reply:
Refer to the test case "Locally this can be run without any additional settings". If we dont have the env variable set which necessarily means its not push and we can run on local settings. The concern is addressed @Mikaayenson

Reply:
Updated summary